components/UploadForm.vue

```diff
@@ -0,0 +1,671 @@
+        @dragleave.prevent="isDragging = false"
+        @click="$refs.imageFile.click()"
+      >
+        <UIcon name="i-heroicons-folder-arrow-down" class="w-5 h-5" />
+        <p class="text-gray-600 my-[16px]" v-text="`把檔案拖到此處上傳或`" />
+        <UButton type="button" variant="ghost" @click="$refs.imageFile.click()">
+          選擇檔案
+        </UButton>
+        <p class="text-xs text-gray-500 mt-2" v-text="`建議檔案大小: < 20 MB`" />
+        <input
+          ref="imageFile"
+          type="file"
+          multiple
+          class="hidden"
+          @change="onFileUpload"
+        >
+      </form>
+
+      <div v-if="fileRecords.length" class="flex flex-col w-full">
+        <table class="w-full">
+          <tbody class="w-full">
+            <tr
+              v-for="(
+                { fileData, fileName, fileSize, fileType }, index
+              ) of fileRecords"
+              :key="fileName"
+              class="flex justify-between items-center border-b-shade-gray border-b-[1px] text-dark-gray hover:bg-light-gray transition-colors w-full"
+            >
+              <td class="py-[4px]">
+                <ImgPreviewer
+                  :file-type="fileType"
+                  :file-data="fileData"
+                  size="small"
+                />
+              </td>
+              <td>
+                <div class="flex flex-col">
+                  <p class="font-semibold text-gray-700">
+                    {{ fileName }}
+                  </p>
+                  <p class="text-gray-500 text-sm">
+                    {{ Math.round(fileSize * 0.001) }} KB
+                  </p>
+                </div>
+              </td>
+              <td>
+                <UIcon
+                  name="i-heroicons-trash"
+                  class="cursor-pointer text-red-500"
+                  @click="handleDeleteFile(index)"
+                />
+              </td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    </div>
+    <div v-if="uploadStatus" class="w-full">
+      <div class="space-y-3">
+        <div class="flex justify-between items-center">
+          <UBadge color="Badge" variant="soft">
+            {{ uploadStatus }}
+          </UBadge>
+          <p class="text-xs text-gray-500">
+            請勿關閉此視窗，直到上傳完成。
+          </p>
+        </div>
+        <UProgress
+          animation="carousel"
+          color="primary"
+          class="w-full"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { storeToRefs } from 'pinia'
+import exifr from 'exifr'
+import ePub from 'epubjs'
+import { BigNumber } from 'bignumber.js'
+import { fileToArrayBuffer, digestFileSHA256, calculateIPFSHash } from '~/utils/index'
+import { useFileUpload } from '~/composables/useFileUpload'
+import {
+  estimateBundlrFilePrice,
+  uploadSingleFileToBundlr
+} from '~/utils/arweave'
+import { sendLIKE } from '~/utils/cosmos' // You'll need to implement this
```

Comment:
```suggestion
import { sendLIKE } from '~/utils/cosmos'
```
Did we implement this?

Reply:
Yes, I was betrayed by ai 🙂